scripts/linera_net_helper.sh

```diff
@@ -9,32 +9,24 @@ function linera_spawn_and_read_wallet_variables() {
     # Clean up the temporary directory in case we exit.
     trap 'rm -rf "$LINERA_TMP_DIR"' EXIT
 
-    LINERA_TMP_PID="$LINERA_TMP_DIR/pid"
     LINERA_TMP_OUT="$LINERA_TMP_DIR/out"
     LINERA_TMP_ERR="$LINERA_TMP_DIR/err"
     mkfifo "$LINERA_TMP_ERR" || exit 1
 
-    (
-        # Ignoring SIGPIPE to keep `tee` alive after `sed` exits below, closing
-        # LINERA_TMP_ERR.
-        trap '' PIPE
-
-        # Start the main process.
-        "$@" 2> >(tee "$LINERA_TMP_ERR" 2>/dev/null) 1>"$LINERA_TMP_OUT" &
-
-        # Remember the PID of the service for later.
-        jobs -p > "$LINERA_TMP_PID"
-    )
-    LINERA_NET_PID=$(cat "$LINERA_TMP_PID")
+    "$@" >"$LINERA_TMP_OUT" 2>"$LINERA_TMP_ERR" &
+    LINERA_NET_PID=$!
 
     # When the shell exits, we will clean up the top-level jobs (if any), the temporary
     # directory, and the main process. Handling future top-level jobs here is useful
     # because calling `trap` again will be replace this handler.
     trap 'jobs -p | xargs -r kill && rm -rf "$LINERA_TMP_DIR" && kill "$LINERA_NET_PID"' EXIT
```

Comment:
Because we removed the subshell line 17-28, we don't need to special case LINERA_NET_PID any more (i.e. it will be listed in `jobs -p` on exit). So this entire line becomes `trap 'jobs -p | xargs -r kill && rm -rf "$LINERA_TMP_DIR"` and can be moved up and replace line 10.

Reply:
Ah yes, absolutely!  I wasn't sure that that was the only reason to special-case `LINERA_NET_PID`.